Sources/ImageFetcher/ImageProcessor.swift

```diff
@@ -27,16 +27,46 @@
 
 import Foundation
 
+/// An object that decompresses and processes image data.
 public final class ImageProcessor: ImageProcessing {
+    private let queue: Queue
+
+    /// Creates an image processor with the given queue and concurrent operation limit.
+    /// - Parameters:
+    ///   - queue: A queue that regulates the execution of operations.
+    ///   - maxConcurrent: The maximum number of queued operations that can run at the same time.
+    public init(
+        queue: Queue = OperationQueue(),
+        maxConcurrent: Int? = 2
```

Comment:
`maxConcurrent` is optional to permit the operation queue to determine this number dynamically based on current system conditions when it is `nil`

Reply:
Could you update the header doc to note this behavior?